libraries/AP_HAL/examples/RCInput/RCInput.cpp

```diff
@@ -34,7 +34,7 @@ void loop(void)
         }
     }
     if (changed) {
-        for (uint8_t i=0; i<max_channels; i++) {
+        for (uint8_t i=0; i<=max_channels; i++) {
```

Comment:
Guludo's variable name is better imo, but I agree with Lucas - one could just use `nchannels`. 

I thought that since this was an example sketch, it might have parallels to the main code where having 2 different variables may be necessary. I have no evidence for this hypothesis though. 

`libraries/AP_HAL/examples/RCInputToRCOutput.cpp` has the same bug as well I believe. It writes PWM values to all channels, but only prints to `nchannels-1`.